pkg/validations/upgradevalidations/immutableFields.go

```diff
@@ -88,24 +88,12 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 		return fmt.Errorf("spec.externalEtcdConfiguration is immutable")
 	}
 
-	if !v1alpha1.RefSliceEqual(nSpec.IdentityProviderRefs, oSpec.IdentityProviderRefs) {
```

Comment:
you will still need to validate `IdentityProviderRefs` mutability

Reply:
Validated